truffle/src/main/java/org/jruby/truffle/nodes/core/EncodingConverterNodes.java

```diff
@@ -11,72 +11,35 @@
 
 import org.jcodings.Encoding;
+import org.jcodings.EncodingDB;
 import org.jcodings.transcode.EConv;
 import org.jcodings.transcode.Transcoder;
 import org.jcodings.transcode.TranscoderDB;
 import org.jruby.Ruby;
 import org.jruby.runtime.ThreadContext;
 import org.jruby.runtime.builtin.IRubyObject;
+import org.jruby.truffle.nodes.dispatch.CallDispatchHeadNode;
+import org.jruby.truffle.nodes.dispatch.DispatchHeadNodeFactory;
 import org.jruby.truffle.runtime.RubyContext;
+import org.jruby.truffle.runtime.UndefinedPlaceholder;
 import org.jruby.truffle.runtime.core.*;
+import org.jruby.truffle.runtime.hash.HashOperations;
+import org.jruby.truffle.runtime.hash.KeyValue;
 import org.jruby.util.ByteList;
 import org.jruby.util.io.EncodingUtils;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @CoreClass(name = "Encoding::Converter")
 public abstract class EncodingConverterNodes {
 
-    @CoreMethod(names = "convpath")
-    public abstract static class ConvPathNode extends CoreMethodNode {
-
-        public ConvPathNode(RubyContext context, SourceSection sourceSection) {
-            super(context, sourceSection);
-        }
-
-        public ConvPathNode(ConvPathNode prev) {
-            super(prev);
-        }
-
-        @TruffleBoundary
-        @Specialization
-        public RubyArray convpath(RubyEncodingConverter converter) {
-            notDesignedForCompilation();
-
-            // Adapated from RubyConverter - see attribution there
-
-            Ruby runtime = getContext().getRuntime();
-
-            EConv ec = converter.getEConv();
-
-            Object[] result = new Object[ec.numTranscoders];
-            int r = 0;
-
-            for (int i = 0; i < ec.numTranscoders; i++) {
-                Transcoder tr = ec.elements[i].transcoding.transcoder;
-                Object v;
-                if (EncodingUtils.DECORATOR_P(tr.getSource(), tr.getDestination())) {
-                    v = new RubyString(getContext().getCoreLibrary().getStringClass(), new ByteList(tr.getDestination()));
-                } else {
-                    Encoding source = runtime.getEncodingService().findEncodingOrAliasEntry(tr.getSource()).getEncoding();
-                    Encoding destination = runtime.getEncodingService().findEncodingOrAliasEntry(tr.getDestination()).getEncoding();
-
-                    v = new RubyArray(getContext().getCoreLibrary().getArrayClass(),
-                            new Object[]{
-                                RubyEncoding.getEncoding(source),
-                                RubyEncoding.getEncoding(destination)
-                            }, 2);
-                }
-                result[r++] = v;
-            }
-
-            return new RubyArray(getContext().getCoreLibrary().getArrayClass(), result, result.length);
-        }
-
-    }
-
-    @CoreMethod(names = "initialize", required = 2)
+    @RubiniusOnly
+    @CoreMethod(names = "initialize_jruby", required = 2, optional = 1)
```

Comment:
make it private :wink: initialize is implicitly private but not when suffixed of course.